src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java

```diff
@@ -291,8 +292,11 @@ static String createValidParamName(String name, int idx) {
         return result;
     }
 
-    static MethodSpec buildFunction(
+    static List<MethodSpec> buildFunction(
```

Comment:
I don't see why returning a list is necessary. All that should be required is that the method is slightly different if payable is set, otherwise the logic should be identical to as per previous.